pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -308,3 +327,45 @@ func (r *CloudConfigReconciler) setDegradedCondition(ctx context.Context) error
 	klog.Info("Cloud Config Controller is degraded")
 	return r.syncStatus(ctx, co, conds)
 }
+
+func (r *CloudConfigReconciler) setUpgradeableFalseCondition(ctx context.Context) error {
+	co, err := r.getOrCreateClusterOperator(ctx)
+	if err != nil {
+		return err
+	}
+
+	conds := []configv1.ClusterOperatorStatusCondition{
+		newClusterOperatorStatusCondition(cloudConfigControllerAvailableCondition, configv1.ConditionTrue, ReasonAsExpected,
+			"Cloud Config Controller works as expected"),
+		newClusterOperatorStatusCondition(cloudConfigControllerDegradedCondition, configv1.ConditionFalse, ReasonAsExpected,
+			"Cloud Config Controller works as expected"),
+		newClusterOperatorStatusCondition(configv1.OperatorUpgradeable, configv1.ConditionFalse, ReasonNotUpgradeable,
+			"Cloud Config Controller is not upgradeable"),
```

Comment:
I wonder if we can add a string to the signature here so that we can decorate this message with a bit more useful information?

Reply:
hmm, making that text more specific doesn't quite feel right, what if the text directs users to check the logs?

Reply:
If the text can be varied by the caller, I think it's ok, but checking logs also works

Reply:
part of thing is, this code will only exist for 4.12.z and only gets engaged for nutanix

if you are cool with adding a comment to look at the logs, i think that makes sense to me